### DISCUSSION TITLE
[C++] Deallocating memory at the end of the onnx inference

### DISCUSSION THREAD

**Discussion by lorenzobattelli** [microsoft/onnxruntime#16499] [General]
I found out that, at the end of my python script which uses my c++ DLL in order to make a onnx inference, the memory acquired is not fully release. (Windows 11). I already provided my code some destructors for the Ort::Session* dynamically allocated. But yet, not all the memory is released nonetheless. How can I assure to free all the memory?

**Comment by Red-Eyed** [comment]
Hello!
Could you provide minimum working code  example?

Try to call `del session` and see what happens
